### PR TITLE
Update scripts to pass shellcheck 0.8

### PR DESCRIPTION
**Pull request by jacob-delgado**
**Please provide a description of this PR:**

Updating shellcheck dependency

### CODE REVIEW

File: release/downloadIstioCandidate.sh [hunk 1]
```diff
@@ -28,14 +28,14 @@ set -e
 
 # Determines the operating system.
 OS="$(uname)"
-if [ "x${OS}" = "xDarwin" ] ; then
+if [ "${OS}" = "Darwin" ] ; then
```
Comment (ericvn) [inline release/downloadIstioCandidate.sh:31]: Interesting that this is now getting flagged to be fixed. I know it's very old school to use the X prefix, and I wouldn't expect OS or version to start with a `-`.

Comment (jacob-delgado) [reply release/downloadIstioCandidate.sh:31]: Ran from my mac

```bash
➜  release git:(update-shellcheck) (⎈ |admin:default) ISTIO_VERSION="" ./downloadIstioCtl.sh

 Downloading istioctl-1.13.2 from https://github.com/istio/istio/releases/download/1.13.2/istioctl-1.13.2-osx.tar.gz ...
istioctl-1.13.2-osx.tar.gz download complete!

Add the istioctl to your path with:
  export PATH=$PATH:$HOME/.istioctl/bin

Begin the Istio pre-installation check by running:
	 istioctl x precheck

Need more information? Visit https://istio.io/docs/reference/commands/istioctl/
➜  release git:(update-shellcheck) (⎈ |admin:default) ISTIO_VERSION="1.13.1" ./downloadIstioCtl.sh

 Downloading istioctl-1.13.1 from https://github.com/istio/istio/releases/download/1.13.1/istioctl-1.13.1-osx.tar.gz ...
istioctl-1.13.1-osx.tar.gz download complete!

Add the istioctl to your path with:
  export PATH=$PATH:$HOME/.istioctl/bin

Begin the Istio pre-installation check by running:
	 istioctl x precheck

Need more information? Visit https://istio.io/docs/reference/commands/istioctl/
```

**Review by howardjohn — APPROVED** [review]

**Pull request merged**